### PR TITLE
dh-make-golang make: directly use the user preferred tag specified in "-git_revision" option

### PR DESCRIPTION
**Pull request by wlyu2**
When a user specifies a valid tag when using

    dh-make-golang make -git_revision {some valid tag}

if there is another tag pointing to the same commit as the specified tag, then dh-make-golang might use the other tag to determine the package version.

This patch modifies the behavior of "dh-make-golang make" such that, if the user specifies a valid tag in "-git_revision" option, "dh-make-golang make" will always use the specified tag to determine the upstream package version.

This patch also trivially changes the tests written in version_test.go so that the tests adjusts to the new function signature of version.go:pkgVersionFromGit().

### CODE REVIEW

**Comment by wlyu2** [comment]
This patch attempts to fix #226 .

File: version.go [hunk 1]
```diff
@@ -32,16 +33,32 @@ var (
...
+	var cmd *exec.Cmd
+
+	// If the user specifies a valid tag as the preferred revision, that tag should be used without additional heuristics.
+	if u.rr != nil {
+		if out, err := u.rr.VCS.Tags(gitdir); err == nil && slices.Contains(out, preferredRev) {
+			latestTag = preferredRev
+			goto FoundLatestTag
+		}
+	}
+
```
Comment (wlyu2) [inline version.go:51]: These are some comments regarding why I use a `goto` statement. Please let me know if there are better alternatives.

I want to use the `git describe` heuristics as a fallback in case I cannot verify if the user specified git revision is valid or not.

The logic I want to implement is:
```
if (u.rr is valid (which makes u.rr.VCS.Tags meaningful) and the user specified tag exists as a valid tag) {
    set the latest tag to this user specified tag
} else {
    the fallback heuristics
}
```

However, to implement the "if" part of the above pseudocode, I cannot find a way to implement it as a single if clause. This will result in the "else" part being written out twice, and I want to avoid such duplication.

In other words, I am trying to deal with the problem in this post: https://stackoverflow.com/questions/13197425/multiple-if-with-one-else

William

**Comment by ottok** [comment]
Does @n-peugnet or @Maytha8 want to review/approve this?

**Review by Maytha8 — APPROVED** [review]

**Comment by wlyu2** [comment]
Closes #226

File: version.go (hunk 1 above)
Comment (NightTsarina) [reply version.go:51]: Wouldn't it be better to set a `foundTag` flag or reusing the `if len(latestTag) > 0` test?

**Review by NightTsarina — APPROVED** [review]
